clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7272,6 +7272,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-fdelayed-template-parsing");
   }
 
+  if (Args.hasFlag(options::OPT_fms_reference_binding,
+                   options::OPT_fno_ms_reference_binding,
+                   IsWindowsMSVC && !HaveCxx20))
```

Comment:
Could we do something better for the non-windows or c++20 cases?

We could emit a warning saying the flag is unsupported, or we could put those conditions first so that we don't call hasFlag, which would make clang warn about the unused option.

Reply:
I am not quite following. I modelled this flag similar to how we do other ms flags such as `-fms-extensions` and `-fdelayed-template-parsing`. That is it is enabled by cl driver options or enabled by default for MSVC targets.
It is still valid for you to explicitly do `-fms-extensions` on non-msvc targets if you want those behaviours for reasons.

When porting some old MSVC only games to certain console manufacturers it was common to use `-fms-extensions` and friends to get it building on their vendor clang :).